packages/qwik/src/lib/ssgoi-transition.tsx

```diff
@@ -0,0 +1,42 @@
+import { component$, Slot, useVisibleTask$, useSignal } from "@builder.io/qwik";
+import { useSsgoi } from "./context";
+import { transition as coreTransition } from "@ssgoi/core";
+
+interface SsgoiTransitionProps {
+  id: string;
+}
+
+export const SsgoiTransition = component$<SsgoiTransitionProps>(({ id }) => {
+  const elementRef = useSignal<HTMLDivElement>();
+
+  useVisibleTask$(({ track }) => {
+    const element = track(() => elementRef.value);
```

Comment:
for signals you can just write `track(elementRef)`, but anyway, the value will never change, and it will always exist in a visible task. So no need to track.

Reply:
actually, not quite true: if you are conditionally rendering a component then the ref will change every time a new DOM element is assigned or removed.